packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs

```diff
@@ -618,8 +618,11 @@ private CodeWriter AppendRaw(ReadOnlySpan<char> span)
         {
             if (span.Length == 0 )
                 return this;
-
-            AddSpaces();
+            // pre-processor directives do not need indentation
```

Comment:
Can we move this into the AddSpaces function to encapsulate all indentation logic in that function?